### PR TITLE
Featured/upgrade java 17 and spring boot 3

### PR DESCRIPTION
**Pull request by michelu89**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -502,34 +475,75 @@
...
+                                <id>build-native</id>
+                                <goals>
+                                    <goal>compile-no-fork</goal>
+                                </goals>
+                                <phase>package</phase>
+                            </execution>
+                        </executions>
                         <configuration>
-                            <mainClass>${main.class}</mainClass>
+                            <mainClass>io.openmanufacturing.ame.Application</mainClass>
```
Comment (atextor) [inline pom.xml:498]: Why do you not use the (still existing) `${main.class}` property any more?

File: src/main/java/io/openmanufacturing/ame/Application.java [hunk 2]
```diff
@@ -13,12 +13,31 @@
...
+      // Since we have no rights with the local service in the class path we have to use the MigratorService ourselves.
+      final MigratorServiceLoader migratorService = new MigratorServiceLoader();
+      final Field migratorServiceRef = ReflectionUtils.findField( MigratorServiceLoader.class, "migratorService" );
+      final Field instanceRef = ReflectionUtils.findField( MigratorServiceLoader.class, "instance" );
+      migratorServiceRef.setAccessible( true );
+      ReflectionUtils.setField( migratorServiceRef, migratorService, new MigratorService() );
+      instanceRef.setAccessible( true );
+      ReflectionUtils.setField( instanceRef, migratorService, migratorService );
+
+      // Spring and GrallVm cannot launch Javascript engines at the moment, so this must be disabled for now.
```
Comment (atextor) [inline src/main/java/io/openmanufacturing/ame/Application.java:38]: GraalVM

File: src/main/java/io/openmanufacturing/ame/Application.java [hunk 3]
```diff
@@ -13,12 +13,31 @@
...
+
+import io.openmanufacturing.sds.aspectmodel.shacl.constraint.JsConstraint;
+import io.openmanufacturing.sds.aspectmodel.versionupdate.MigratorService;
+import io.openmanufacturing.sds.aspectmodel.versionupdate.MigratorServiceLoader;
 
 @SpringBootApplication
 public class Application {
    public static void main( final String[] args ) {
+      // Since we have no rights with the local service in the class path we have to use the MigratorService ourselves.
+      final MigratorServiceLoader migratorService = new MigratorServiceLoader();
```
Comment (atextor) [inline src/main/java/io/openmanufacturing/ame/Application.java:30]: Such workarounds should not be necesary for other ESMF modules. Please consider opening an issue to change the MigratorServiceLoader instead.

**Review by atextor — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 4]
```diff
@@ -819,15 +819,12 @@
...
-            <id>github</id>
-            <name>OpenManufacturingPlatform SDS-SDK</name>
-            <url>https://maven.pkg.github.com/OpenManufacturingPlatform/sds-sdk</url>
+            <id>esmf-sdk-release</id>
+            <name>ESMF-SDK</name>
+            <url>https://maven.pkg.github.com/eclipse-esmf/esmf-sdk</url>
             <releases>
                 <enabled>true</enabled>
             </releases>
-            <snapshots>
```
[the remark this answers is not in the document]
Comment (michelu89) [reply pom.xml]: Reverted

File: src/main/java/io/openmanufacturing/ame/Application.java (hunk 3 above)
Comment (michelu89) [reply src/main/java/io/openmanufacturing/ame/Application.java:30]: Fixed

File: src/main/java/io/openmanufacturing/ame/Application.java (hunk 2 above)
Comment (michelu89) [reply src/main/java/io/openmanufacturing/ame/Application.java]: Fixed

**Review by atextor — APPROVED** [review]

**Pull request merged**